### ISSUE TITLE
Game Crashes on Stage Start Each Time

### ISSUE THREAD

**Issue by gRally-Issues** [gRally/dev#595]
# User: Scout
---

#### What happened?
The game crashes with an access violation error.

#### Steps to reproduce:
Start the game.

#### Considerations...

[gSim](https://www.grally.net/u/uploads/4cec95b4a4c9f62202e0ec0ca485efc7433ebbf1.log)

### [log](https://www.grally.net/u/uploads/485712e95d90c996fa103525146dc4ae26df792f.zip)

`v1.0.3.1`

**Comment by TommyJ1994** [comment]
In `C:\Users\user\AppData\LocalLow\ghiboz.com\gRallySim\Crashes\Crash_2018-05-27_170238`

```
gRallySim [version: Unity 2017.4.1f1 (9231f953d9d3)]

gSim_64.dll caused an Access Violation (0xc0000005)
  in module gSim_64.dll at 0033:3955b75c.

Error occurred at 2018-05-27_170302.
C:\Program Files (x86)\Steam\steamapps\common\gRally\gRallySim.exe, run by user.
81% memory in use.
8144 MB physical memory [1532 MB free].
21082 MB paging file [2757 MB free].
134217728 MB user address space [134209015 MB free].
Read from location 00000024 caused an access violation.

Context:
RDI:    0x265ff528  RSI: 0x881d94b0  RAX:   0x00000000
RBX:    0x265ff4e0  RCX: 0x00000004  RDX:   0x00000002
RIP:    0x3955b75c  RBP: 0x265ff561  SegCs: 0x00000033
EFlags: 0x00010202  RSP: 0x265ff450  SegSs: 0x0000002b
R8:    0x00000001  R9: 0x61d70000  R10:   0x265ff648
R11:    0x265ff528  R12: 0x00000000  R13:   0x5c57a9c4
R14:    0x00000008  R15: 0x00000000

Bytes at CS:EIP:
8b 5c c8 04 48 8b cf e8 b8 1e fa ff 8b c3 48 8b 

Stack:
0x265ff450: 015f4a2d 00000722 13130000 00000000 -J_."...........
0x265ff460: 00000008 00000000 3955ae4f 00007ffd ........O.U9....
0x265ff470: fffffffe ffffffff 265ff528 0000002f ........(._&/...
0x265ff480: 7ee70d07 00009f7b 13130000 00000000 ...~{...........
0x265ff490: 265ff648 0000002f 3955d81b 00007ffd H._&/.....U9....
0x265ff4a0: 265ff4e0 0000002f 00000000 ffffffe6 .._&/...........
0x265ff4b0: 265ff4e0 0000002f 13130000 00000000 .._&/...........
0x265ff4c0: 00000000 ffffffe8 265ff4e0 0000002f .........._&/...
0x265ff4d0: 13130000 00000000 fffffffe ffffffff ................
0x265ff4e0: 65636341 6172656c 13006574 00000000 Accelerate......
0x265ff4f0: 0000000a 00000000 0000000f 00000000 ................
0x265ff500: 00000001 00000000 00000002 46fffe00 ...............F
0x265ff510: c7000000 3f800000 00000000 00000001 .......?........
0x265ff520: 40d00000 fffffff0 65636341 6172656c ...@....Accelera
0x265ff530: 13006574 00000000 0000000a 00000000 te..............
0x265ff540: 0000000f 00000000 00000001 00000000 ................
0x265ff550: 00000002 46fffe00 c7000000 3f800000 .......F.......?
0x265ff560: 00000000 00000001 40d00000 fffffff7 ...........@....
0x265ff570: 265ff648 0000002f 7ee70df7 00009f7b H._&/......~{...
0x265ff580: 00000000 00000000 00000000 00000000 ................
0x265ff590: 3ba3d70a 00000000 00000000 00000000 ...;............
0x265ff5a0: 265ff7a8 0000002f 881d94b0 000001e5 .._&/...........
0x265ff5b0: 265ff6c0 0000002f 3955bab2 00007ffd .._&/.....U9....
0x265ff5c0: 0000000f 00000000 00000000 00000000 ................
0x265ff5d0: 0000000f 00000000 13130000 00000000 ................
0x265ff5e0: 00000000 00000002 015f4a9a 00000736 .........J_.6...
0x265ff5f0: 13130000 00000000 00000000 00000006 ................
0x265ff600: 015f4aaa 00000738 13130000 00000000 .J_.8...........
0x265ff610: 00000000 0000000a 015f4ab9 00000739 .........J_.9...
0x265ff620: 13130000 00000000 00000000 0000000f ................
0x265ff630: 015f4ac9 0000073a 13130000 00000000 .J_.:...........
0x265ff640: 00000000 00000013 65636341 6172656c ........Accelera
0x265ff650: 13006574 00000000 0000000a 00000000 te..............
0x265ff660: 0000000f 00000000 00000001 00000000 ................
0x265ff670: 00000002 46fffe00 c7000000 3f800000 .......F.......?
0x265ff680: 00000000 00000001 40d00000 00000020 ...........@ ...
0x265ff690: fffffffe ffffffff 13130000 00000000 ................
0x265ff6a0: 00000000 00000024 015f4ae8 00000740 ....$....J_.@...
0x265ff6b0: 13130000 00000000 00000000 00000028 ............(...
0x265ff6c0: 015f4ae8 00000741 13130000 00000000 .J_.A...........
0x265ff6d0: 00000000 0000002d 015f4af8 00000742 ....-....J_.B...
0x265ff6e0: 13130000 00000000 00000000 00000031 ............1...
0x265ff6f0: 265ff7a8 0000002f 7ee70c97 00009f7b .._&/......~{...
0x265ff700: 3ba3d70a 00000000 00000000 00000000 ...;............
0x265ff710: 00000000 00000000 00000000 00000000 ................
0x265ff720: 015f4b07 00000745 00000000 00000000 .K_.E...........
0x265ff730: 00000000 00000000 5c57a9c4 00000005 ..........W\....
0x265ff740: 00000000 00000000 13eb3d10 000001e6 .........=......
0x265ff750: 13eb3550 000001e6 13eb3550 000001e6 P5......P5......
0x265ff760: 265ff809 0000002f 3950ed7e 00007ffd .._&/...~.P9....
0x265ff770: 13eb3550 000001e6 13eb3550 000001e6 P5......P5......
0x265ff780: 13eb3d10 000001e6 13130000 00000000 .=..............
0x265ff790: 00000000 0000003c 015f4c30 00000751 ....<...0L_.Q...
0x265ff7a0: fffffffe ffffffff 6f726854 656c7474 ........Throttle
0x265ff7b0: 015f4c00 00000753 00000008 00000000 .L_.S...........
0x265ff7c0: 0000000f 00000000 6f726854 656c7474 ........Throttle
0x265ff7d0: 13130000 00000000 00000008 00000000 ................
0x265ff7e0: 0000000f 00000000 7ee70e27 00009f7b ........'..~{...
0x265ff7f0: 00000000 00000000 00000000 00000000 ................
0x265ff800: 00000000 00000000 00000000 00000000 ................
0x265ff810: 447a0000 00000000 00000000 00000000 ..zD............
0x265ff820: 4e6e6b28 00000000 00000000 00000000 (knN............
0x265ff830: 00000000 00000000 c030c2e0 000001e5 ..........0.....
0x265ff840: 5c57a9c4 00000005 00000000 00000000 ..W\............
0x265ff850: 00000004 00000000 13eb3550 000001e6 ........P5......
0x265ff860: 265ffa20 0000002f 3954cc05 00007ffd  ._&/.....T9....
0x265ff870: 13eb3550 000001e6 00000000 00000000 P5..............
0x265ff880: 13eb3550 000001e6 265ffa10 0000002f P5........_&/...
0x265ff890: fffffffe ffffffff 39504641 00007ffd ........AFP9....
0x265ff8a0: b888450e 00000010 c02f3c00 000001e5 .E.......</.....
0x265ff8b0: 3968d160 00007ffd 265ffa10 0000002f `.h9......_&/...
0x265ff8c0: fffffffe ffffffff c02f3c00 000001e5 .........</.....
0x265ff8d0: fffffffe ffffffff d83ce7d0 000001e5 ..........<.....
0x265ff8e0: 00000001 00000000 39504483 00007ffd .........DP9....
0x265ff8f0: 7ee701f7 00009f7b 7ee701c7 00009f7b ...~{......~{...
0x265ff900: c030c2e0 000001e5 13eb3550 000001e6 ..0.....P5......
0x265ff910: 00000004 00000000 3954760b 00007ffd .........vT9....
0x265ff920: 0000001e 00000000 0000001e 00000000 ................
0x265ff930: 0000001e 00000000 3961d468 00007ffd ........h.a9....
0x265ff940: 265ff9d0 00000000 00000000 0000002f .._&......../...
0x265ff950: 00000000 00000000 00000000 00000000 ................
0x265ff960: 00000000 00000000 00000014 00000000 ................
0x265ff970: 00000000 00000000 00000000 00000000 ................
0x265ff980: 00000000 00000000 00001388 00000000 ................
0x265ff990: 00000000 00000000 3961d468 00007ffd ........h.a9....
0x265ff9a0: 265f0080 0000002f 39624380 00007ffd .._&/....Cb9....
0x265ff9b0: 000001b0 00000000 396244b8 00007ffd .........Db9....
0x265ff9c0: 00000000 0000959f d83ce780 000001e5 ..........<.....
0x265ff9d0: 00000000 0000959f d83ce780 000001e5 ..........<.....
0x265ff9e0: 3961d458 00007ffd 92410002 000001e5 X.a9......A.....
0x265ff9f0: 00000000 00000000 00000000 00000000 ................
0x265ffa00: 00000000 00000000 fffffffe ffffffff ................
0x265ffa10: 61647055 6e496574 73747570 00000100 UpdateInputs....
0x265ffa20: 0000000c 00000000 0000000f 00000000 ................
0x265ffa30: 00000000 00000000 00000000 00000000 ................
0x265ffa40: 00000000 00000000 00000000 00000000 ................
0x265ffa50: 00000000 00000000 00000000 00000000 ................
0x265ffa60: 00000000 00000000 00000000 00000000 ................
0x265ffa70: 00000000 00000000 00000000 00000000 ................
0x265ffa80: 00000000 00000000 00000000 00000000 ................
0x265ffa90: 00000000 00000000 00000000 00000000 ................
0x265ffaa0: 00000000 00000000 00000000 00000000 ................
0x265ffab0: 00000000 00000000 00000000 00000000 ................
0x265ffac0: 00000000 00000000 00000000 00000000 ................
0x265ffad0: 00000000 00000000 00000000 00000000 ................
0x265ffae0: 00000000 00000000 00000000 00000000 ................
0x265ffaf0: 00000000 00000000 00000000 00000000 ................
0x265ffb00: 00000000 00000000 00000000 00000000 ................
0x265ffb10: 00000000 00000000 00000000 00000000 ................
0x265ffb20: 00000000 00000000 00000000 00000000 ................
0x265ffb30: 00000000 00000000 00000000 00000000 ................
0x265ffb40: 00000000 00000000 00000000 00000000 ................
0x265ffb50: 00000000 00000000 00000000 00000000 ................
0x265ffb60: 00000000 00000000 00000000 00000000 ................
0x265ffb70: 00000000 00000000 00000000 00000000 ................
0x265ffb80: 00000000 00000000 00000000 00000000 ................
0x265ffb90: 00000000 00000000 00000000 00000000 ................
0x265ffba0: 00000000 00000000 00000000 00000000 ................
0x265ffbb0: 00000000 00000000 00000000 00000000 ................
0x265ffbc0: 00000000 00000000 00000000 00000000 ................
0x265ffbd0: 00000000 00000000 00000000 00000000 ................
0x265ffbe0: 00000000 00000000 00000000 00000000 ................
0x265ffbf0: 00000000 00000000 00000000 00000000 ................
0x265ffc00: 00000000 00000000 00000000 00000000 ................
0x265ffc10: 00000000 00000000 00000000 00000000 ................
0x265ffc20: 00000000 00000000 00000000 00000000 ................
0x265ffc30: 00000000 00000000 00000000 00000000 ................
0x265ffc40: 00000000 00000000 00000000 00000000 ................
0x265ffc50: 00000000 00000000 00000000 00000000 ................
0x265ffc60: 00000000 00000000 00000000 00000000 ................
0x265ffc70: 00000000 00000000 00000000 00000000 ................
0x265ffc80: 00000000 00000000 00000000 00000000 ................
0x265ffc90: 00000000 00000000 00000000 00000000 ................
0x265ffca0: 00000000 00000000 00000000 00000000 ................
0x265ffcb0: 00000000 00000000 00000000 00000000 ................
0x265ffcc0: 00000000 00000000 00000000 00000000 ................
0x265ffcd0: 00000000 00000000 00000000 00000000 ................
0x265ffce0: 00000000 00000000 00000000 00000000 ................
0x265ffcf0: 00000000 00000000 00000000 00000000 ................
0x265ffd00: 7ee70077 00009f7b 35374342 32344633 w..~{...BC753F42
0x265ffd10: 00000000 00000000 00000000 00000000 ................
0x265ffd20: 00000000 00000000 00000000 00000000 ................
0x265ffd30: 00000000 00000000 00000000 00000000 ................
0x265ffd40: 00000000 00000000 00000000 00000000 ................
0x265ffd50: 00000000 00000000 3960bfb3 00007ffd ..........`9....
0x265ffd60: af2a3ee0 000001e5 af2a3ee0 000001e5 .>*......>*.....
0x265ffd70: 00000000 00000000 00000000 00000000 ................
0x265ffd80: fffffffe ffffffff 00000000 00000000 ................
0x265ffd90: 144838c0 000001e6 6f4ec1ae 00007ffd .8H.......No....
0x265ffda0: 144838c0 000001e6 00000000 00000000 .8H.............
0x265ffdb0: 00000000 00000000 00000000 00000000 ................
0x265ffdc0: 00000000 00000000 6fc83034 00007ffd ........40.o....
0x265ffdd0: 00000000 00000000 00000000 00000000 ................
0x265ffde0: 00000000 00000000 00000000 00000000 ................
0x265ffdf0: 00000000 00000000 724f1551 00007ffd ........Q.Or....
0x265ffe00: 00000000 00000000 00000000 00000000 ................
0x265ffe10: 00000000 00000000 00000000 00000000 ................
0x265ffe20: 00000000 00000000 6f6bb940 00007ffd ........@.ko....
0x265ffe30: 265fe5d0 0000002f 265fe5d0 0000002f .._&/....._&/...
0x265ffe40: 00000000 00000000 00000000 00000000 ................
0x265ffe50: 00000000 00000000 00000000 00000000 ................
0x265ffe60: 00000000 00000000 00000000 00000000 ................
0x265ffe70: 00000000 00000000 00000000 00000000 ................
0x265ffe80: 00000000 00000000 00000000 00000000 ................
0x265ffe90: 00000000 00000000 00000000 00000000 ................
0x265ffea0: 00000000 00000000 00000000 00000000 ................
0x265ffeb0: 00000000 00000000 00000000 00000000 ................
0x265ffec0: 00000000 00000000 00000000 00000000 ................
0x265ffed0: 00000000 00000000 00000000 00000000 ................
0x265ffee0: 00000000 00000000 00000000 00000000 ................
0x265ffef0: 00000000 00000000 00000000 00000000 ................
0x265fff00: 00000000 00000000 00000000 00000000 ................
0x265fff10: 00000000 00000000 00000000 00000000 ................
0x265fff20: 00000000 00000000 00000000 00000000 ................
0x265fff30: 00000000 00000000 00000000 00000000 ................
0x265fff40: 00000000 00000000 00000000 00000000 ................
0x265fff50: 00000000 00000000 00000000 00000000 ................
0x265fff60: 00000000 00000000 00000000 00000000 ................
0x265fff70: 00000000 00000000 00000000 00000000 ................
0x265fff80: 00000000 00000000 00000000 00000000 ................
0x265fff90: 00000000 00000000 00000000 00000000 ................
0x265fffa0: 00000000 00000000 00000000 00000000 ................
0x265fffb0: 00000000 00000000 00000000 00000000 ................
0x265fffc0: 00000000 00000000 00000000 00000000 ................
0x265fffd0: 00000000 00000000 00000000 00000000 ................
0x265fffe0: 00000000 00000000 00000000 00000000 ................
0x265ffff0: 00000000 00000000 00000000 00000000 ................

Module 1
C:\WINDOWS\SYSTEM32\xinput1_3.dll
Image Base: 0x00400000  Image Size: 0x0001e000
File Size:  107368      File Time:  2007-04-04_195422
Version:
   Company:    Microsoft Corporation
   Product:    Microsoft® DirectX for Windows®
   FileDesc:   Microsoft Common Controller API
   FileVer:    9.18.944.0
   ProdVer:    9.18.944.0

Module 2
C:\Program Files (x86)\Steam\tier0_s64.dll
Image Base: 0x55f10000  Image Size: 0x00172000
File Size:  379168      File Time:  2018-05-19_000134
Version:
   Company:    Valve Corporation
   Product:     tier0_s Dynamic Link Library
   FileDesc:   tier0_s Dynamic Link Library
   FileVer:    4.51.98.14
   ProdVer:    1.0.0.1

Module 3
C:\Program Files (x86)\Steam\steamclient64.dll
Image Base: 0x56090000  Image Size: 0x00e45000
File Size:  14717504    File Time:  2018-05-19_000134
Version:
   Company:    Valve Corporation
   Product:    Steam
   FileDesc:   Steamclient.dll
   FileVer:    4.51.98.14
   ProdVer:    3.0.0.1

Module 4
C:\Program Files (x86)\Steam\vstdlib_s64.dll
Image Base: 0x57840000  Image Size: 0x00079000
File Size:  477984      File Time:  2018-05-19_000134
Version:
   Company:    Valve Corporation
   Product:    Steam
   FileDesc:   vstdlib_ s.dll
   FileVer:    4.51.98.14
   ProdVer:    3.0.0.1

Module 5
C:\Program Files (x86)\Steam\steamapps\common\gRally\steam_api64.dll
Image Base: 0x578c0000  Image Size: 0x00040000
File Size:  245536      File Time:  2018-05-21_181104
Version:
   Company:    Valve Corporation
   Product:    Steam Client API
   FileDesc:   Steam Client API
   FileVer:    3.92.72.58
   ProdVer:    1.0.0.1


== [end of error.log] ==

```

**Comment by ghiboz** [comment]
fixed in v1.0.6.3
